File: sdk-java/src/test/java/io/superdurable/dex/integ/StateOptionsOverrideTest.java
/*
 * Portions of this file are derived from indeedeng/iwf-java-sdk.
 * Those portions are licensed under the Apache License, Version 2.0.
 * See LICENSES/Apache-2.0.txt and LEGACY_NOTICES.md.
 *
 * Modifications Copyright (c) 2026 Super Durable, Inc.
 *
 * Modifications are licensed under the Super Durable Source License 1.0.
 * Third-Party Materials remain under the Apache License, Version 2.0.
 * See LICENSE and LEGACY_NOTICES.md.
 */

package io.superdurable.dex.integ;

import io.superdurable.dex.Client;
import io.superdurable.dex.testing.DexDevTestEnvironment;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.io.TempDir;

import java.nio.file.Path;
import java.time.Duration;
import java.util.UUID;

import static org.junit.jupiter.api.Assertions.assertEquals;

@Tag("dex-dev")
public final class StateOptionsOverrideTest {
    private static final StateOptionsOverrideWorkflow WORKFLOW =
            new StateOptionsOverrideWorkflow();

    @TempDir
    Path cacheDirectory;

    @Test
    void testStateOptionsOverrideWorkflow() throws Exception {
        try (DexDevTestEnvironment environment = DexDevTestEnvironment.start(
                cacheDirectory,
                WORKFLOW)) {
            final String flowId = "state-options-override-" + UUID.randomUUID();
            environment.client().startFlow(WORKFLOW, flowId, "input");
            assertEquals(
                    "input_state1_start_state1_decide_state2_start_state2_decide",
                    environment.client().waitForFlow(flowId, Duration.ofSeconds(30)).getSingleOutput(String.class));
        }
    }

    void compileMovementOptionsOverride(final Client client) {
        client.startFlow(WORKFLOW, "options-override", "input");
        final String output = client.waitForFlow("options-override").getSingleOutput(String.class);
        consume(output);
    }

    private static void consume(final Object value) {
    }
}

File: sdk-rust/crates/dex-sdk/tests/integ/state_options_override_test.rs
// Portions of this file are derived from indeedeng/iwf-java-sdk.
// Those portions are licensed under the Apache License, Version 2.0.
// See LICENSES/Apache-2.0.txt and LEGACY_NOTICES.md.
//
// Modifications Copyright (c) 2026 Super Durable, Inc.
//
// Modifications are licensed under the Super Durable Source License 1.0.
// Third-Party Materials remain under the Apache License, Version 2.0.
// See LICENSE and LEGACY_NOTICES.md.

use std::time::Duration;

use dex_sdk::{Client, Registry, SdkResult};

use crate::state_options_override_workflow::StateOptionsOverrideWorkflow;
use crate::support::{DexDevTestEnvironment, flow_id};

#[test]
#[ignore = "requires dexcli dev"]
fn test_state_options_override_workflow() {
    let environment =
        DexDevTestEnvironment::start(Registry::new().register(StateOptionsOverrideWorkflow::new()));
    let workflow = StateOptionsOverrideWorkflow::new();
    let flow_id = flow_id("state-options-override");
    environment
        .client
        .start_flow(&workflow, &flow_id, "input".to_string())
        .expect("start options-override Flow");
    assert_eq!(
        "input_state1_start_state1_decide_state2_start_state2_decide",
        environment
            .client
            .wait_for_flow_with_timeout(&flow_id, Duration::from_secs(30))
            .and_then(|result| result.single_output::<String>())
            .expect("complete options-override Flow")
    );
}

#[allow(dead_code)]
fn compile_movement_options_override(client: &Client) -> SdkResult<()> {
    let workflow = StateOptionsOverrideWorkflow::new();
    client.start_flow(&workflow, "options-override", "input".to_string())?;
    let output: String = client.wait_for_flow("options-override")?.single_output()?;
    drop(output);
    Ok(())
}
